app/views/taxons/accordion.html.erb

```diff
@@ -18,33 +19,31 @@
   <div class="grid-row child-topic-contents">
     <div class="column-two-thirds">
       <div class="topic-content">
-
-        <div data-module="accordion-with-descriptions" class="js-hidden">
-          <div class="subsection-wrapper">
-
-            <% presented_taxon.accordion_content.each_with_index do |taxon, index| %>
-              <div class="subsection js-subsection" id="<%= taxon.base_path %>" data-track-count="accordionSection">
-                <div class="subsection-header js-subsection-header">
-                  <h2 class="subsection-title js-subsection-title"><%= taxon.title %></h2>
-                  <% if taxon.description.present? %>
-                    <p class="subsection-description"><%= taxon.description %></p>
-                  <% end %>
-                </div>
-
-                <div class="subsection-content js-subsection-content" id="subsection_content_<%= index + 1 %>">
-                  <% if taxon.can_subscribe? %>
-                    <%= render partial: 'email_alerts', locals: { presented_taxon: taxon } %>
-                  <% end %>
-                  <%= render partial: 'content_list_for_child_taxon', locals: {
-                      section_index: index,
-                      tagged_content: taxon.tagged_content,
-                  } %>
-                </div>
-              </div>
+        <%
+          accordion_contents = []
+        %>
+        <% presented_taxon.accordion_content.each_with_index do |taxon, index| %>
+          <% contents = capture do %>
```

Comment:
I personally find this `capture {}` approach a little inside-out. I think the way I would have expected to use this component is along the lines of:

```erb
accordion_content = [
  { title: "Title", description: "Description", arbitrary_field: "Foo", ... },
]

<%= render "components/accordion", sections: accordion_content do |accordion_section, index| %>
  <!-- Render stuff for the "panel" here, where we have access to the accordion section's hash -->
  <!-- This could even be delegated by rendering a different partial/component! -->
  <div>
    <%= accordion_section[:arbitrary_field] %>
  </div>
<% end %>
```

Reply:
I really like this, I didn't know it was possible to do this, only thought it was possible to yield a passed block.

Reply:
I haven't actually looked into the internals again, but I think I had a branch once upon a time that did something like this (way back when we first looked into turning this into a component). I'm pretty sure it should work, though - you'd just have to have an enumerator in the partial, which calls `yield` for each accordion section.

Reply:
Alec has done the work on this now and I'm going to include it in a future PR - https://github.com/alphagov/collections/compare/accordion-yield?expand=1